adapters/repos/db/vector/hnsw/maintenance.go

```diff
@@ -42,14 +42,11 @@ func (h *hnsw) growIndexToAccomodateNode(id uint64, logger logrus.FieldLogger) e
 
 	// lock h.nodes' individual elements to avoid race between writing to elements
 	// and copying entire slice in growIndexToAccomodateNode method
-	newIndex, err := func() ([]*vertex, error) {
-		h.shardedNodeLocks.RLockAll()
-		defer h.shardedNodeLocks.RUnlockAll()
+	h.shardedNodeLocks.LockAll()
```

Comment:
we are now locking a lot more code (including lines from 64 to 68) that's OK? yes?